modules/tinymce/src/plugins/autosave/main/ts/core/Storage.ts

```diff
@@ -13,13 +13,18 @@ import * as Events from '../api/Events';
 import * as Settings from '../api/Settings';
 
 const isEmpty = (editor: Editor, html?: string) => {
-  const forcedRootBlockName = editor.settings.forced_root_block;
-
-  html = Tools.trim(typeof html === 'undefined' ? editor.getBody().innerHTML : html);
-
-  return html === '' || new RegExp(
-    '^<' + forcedRootBlockName + '[^>]*>((\u00a0|&nbsp;|[ \t]|<br[^>]*>)+?|)<\/' + forcedRootBlockName + '>|<br>$', 'i'
-  ).test(html);
+  if(typeof html === 'undefined') {
+    return editor.dom.isEmpty(editor.getBody());
+  } else {
+    html = Tools.trim(html);
+
+    if(html === '') {
+      return true;
+    } else {
+      const fragment = new DOMParser().parseFromString(html, 'text/html');
+      return editor.dom.isEmpty(fragment as any);
```

Comment:
No casting as `any` here please (either fix why this was needed, or at the very least use a more specific type than any). Also, why did you need to cast this as `any` anyways?

Reply:
Because DOMParser (Katamari) was not DOMParser (default), so when using DOMParser (default) it complained due to not being DOMParser (Katamari), but still worked just fine.

Reply:
I'm confused... there is no `DOMParser` module in `katamari`.

Reply:
Err, yes, dom-globals. I was still having Type on my mind when I wrote and mixed up the modules. The rest was correct though.